lib/resources/dstu2/soarian_device.yaml

```diff
@@ -0,0 +1,32 @@
+---
+name: Device
+field_name_base_url: http://hl7.org/fhir/DSTU2/device-definitions.html#Device
+fields:
+- name: identifier
+  required: 'No'
+  cardinality: 0..*
+  type: Identifier
+  description: Identifier(s) from manufacturer, owner and others.
+  children:
+
+  - name: type
+    required: 'No'
+    cardinality: 0..1
+    type: CodeableConcept
+    description: The type for the identifier that can be used to determine which identifier to use for a specific purpose.
+    url: http://hl7.org/fhir/DSTU2/device-definitions.html#Device.identifier
+    binding:
+      description: The type used to determine which identifier to use for a specific purpose.
+      note: Currently will return SNO code from the below system.
+      terminology:
+      - display: Identifier Type Codes
+        system: http://hl7.org/fhir/identifier-type
```

Comment:
No, In the current implementation it was not returning the identifier type. JIRA has been raised to fix this in a code and we have updated the documentation with appropriate identifier type.